include/quic/gnutls_crypto.hpp

```diff
@@ -69,7 +77,10 @@ namespace oxen::quic
             else
             {
                 path = NULL;
-                mem = {(uint8_t*)input.data(), (uint8_t)input.size()};
+
+                // uint32_t cast to appease narrowing conversion gods,
+                // if cert size won't fit in 32 bits we have bigger problems
+                mem = {(uint8_t*)input.c_str(), (uint32_t)input.size()};
```

Comment:
```suggestion
                mem = {reinterpret_cast<uint8_t*>(input.c_str()), static_cast<uint32_t>(input.size())};
```

Reply:
so gnutls_datum_t is annoying and either requires a copy here into an allocated c-string or something *or* requires casting away const.  For simplicity, as it appears safe, this casts away const.  Could do that in the C++ way though,  I suppose.